File: flutterproject2/lib/pages/Homepage/home_page.dart
import 'package:flutter/material.dart';

class HomePage extends StatelessWidget {
  const HomePage({super.key});

  @override
  Widget build(BuildContext context) {
    // List of new branch images and details
    final List<Map<String, String>> branches = [
      {
        'title': 'Grand Opening - Opol Branch',
        'image': 'assets/branchopol.png',
        'description': 'Join us in celebrating the grand opening of our Opol branch. Enjoy our 3-for-100 promo!',
      },
      {
        'title': 'Outreach Program - Dasmariñas',
        'image': 'assets/branchdasmarinas.png',
        'description': 'Our Dasmariñas branch conducted an outreach program to give back to the community.',
      },
      {
        'title': 'Brewing Soon - Buhisan Branch',
        'image': 'assets/branchbuhisan.png',
        'description': 'Get ready, coffee lovers! Buhisan branch is opening soon!',
      },
      {
        'title': 'Grand Opening - Tabunok Branch',
        'image': 'assets/branchtabunok.png',
        'description': 'Grand opening celebration with special guest Ariel Alegado. Don’t miss out!',
      },
      {
        'title': 'On-the-Go Booth',
        'image': 'assets/branchbooth.png',
        'description': 'Catch us at our on-the-go booths for a quick coffee fix.',
      },
      {
        'title': 'Now Open - PSU Lingayen',
        'image': 'assets/branchlingayen.png',
        'description': 'The PSU Lingayen branch is now open! See you, Kape-Pol!',
      },
    ];

    return Scaffold(
      appBar: AppBar(
        title: const Text(
          'Don Macchiatos ',
          style: TextStyle(fontSize: 22, fontWeight: FontWeight.bold),
        ),
        backgroundColor: Colors.brown,
        centerTitle: true,
      ),
      backgroundColor: Colors.brown,
      drawer: Drawer(
        child: ListView(
          padding: EdgeInsets.zero,
          children: <Widget>[
            const DrawerHeader(
              decoration: BoxDecoration(
                color: Colors.blue,
              ),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: <Widget>[
                  CircleAvatar(
                    radius: 40.0,
                    backgroundImage: AssetImage('assets/donmac.jpg'),
                  ),
                  SizedBox(height: 16.0),
                  Text(
                    'Admin',
                    style: TextStyle(
                      color: Colors.white,
                      fontSize: 20.0,
                      fontWeight: FontWeight.bold,
                    ),
                  ),
                ],
              ),
            ),
            ListTile(
              leading: const Icon(Icons.home),
              title: const Text('Dashboard'),
              onTap: () {
                Navigator.pushNamed(context, '/dashboardpage');
              },
            ),
            ListTile(
              leading: const Icon(Icons.production_quantity_limits_sharp),
              title: const Text('Products'),
              onTap: () {
                Navigator.pushNamed(context, '/products');
              },
            ),
            // ListTile(
            //   leading: const Icon(Icons.storage),
            //   title: const Text('Stocks'),
            //   onTap: () {
            //     Navigator.pushNamed(context, '/stockpage');
            //   },
            // ),
            ListTile(
              leading: const Icon(Icons.bar_chart),
              title: const Text('Sale'),
              onTap: () {
                Navigator.pushNamed(context, '/salespage');
              },
            ),
            ListTile(
              leading: const Icon(Icons.receipt),
              title: const Text('Transaction'),
              onTap: () {
                Navigator.pushNamed(context, '/transactionpage');
              },
            ),
            ListTile(
              leading: const Icon(Icons.verified_user),
              title: const Text('Create User'),
              onTap: () {
                Navigator.pushNamed(context, '/createuserpage');
              },
            ),
            ListTile(
              leading: const Icon(Icons.settings),
              title: const Text('Settings'),
              onTap: () {
                Navigator.pushNamed(context, '/settings');
              },
            ),
            ListTile(
              leading: const Icon(Icons.logout),
              title: const Text('Logout'),
              onTap: () {
                Navigator.pushNamed(context, '/login');
              },
            ),
          ],
        ),
      ),
      body: Padding(
        padding: const EdgeInsets.all(16.0),
        child: SingleChildScrollView(
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              const Text(
                'Branch Highlights',
                style: TextStyle(fontSize: 20, fontWeight: FontWeight.bold),
              ),
              const SizedBox(height: 16.0),
              ListView.builder(
                  shrinkWrap: true,
                  physics: const NeverScrollableScrollPhysics(),
                  itemCount: branches.length,
                  itemBuilder: (context, index) {
                    final branch = branches[index];
                    return Card(
                      elevation: 5.0,
                      margin: const EdgeInsets.only(bottom: 16.0),
                      shape: RoundedRectangleBorder(
                        borderRadius: BorderRadius.circular(10.0),
                      ),
                      child: Column(
                        crossAxisAlignment: CrossAxisAlignment.start,
                        children: [
                          ClipRRect(
                            borderRadius: const BorderRadius.vertical(
                              top: Radius.circular(10.0),
                            ),
                            child: Image.asset(
                              height: MediaQuery.of(context).size.width / 1.5,
                              branch['image']!,
                              width: double.infinity,
                              fit: BoxFit.contain, // Ensures the full image is visible
                            ),
                          ),
                          Padding(
                            padding: const EdgeInsets.all(8.0),
                            child: Text(
                              branch['title']!,
                              style: const TextStyle(
                                fontSize: 18,
                                fontWeight: FontWeight.bold,
                              ),
                            ),
                          ),
                          Padding(
                            padding: const EdgeInsets.symmetric(horizontal: 8.0),
                            child: Text(
                              branch['description']!,
                              style: const TextStyle(fontSize: 14),
                            ),
                          ),
                          const SizedBox(height: 8.0),
                        ],
                      ),
                    );
                  },
                ),
            ],
          ),
        ),
      ),
    );
  }
}

File: web/pages/homepage.js
import Sidebar from "/components/Sidebar";
import React from "react";
import Image from "next/image";

export default function Homepage() {
  return (
    <div style={styles.container}>
      <Sidebar />
      <main style={styles.mainContent}>
        <h1 style={styles.title}>Don Macchiatos</h1>
        <h2 style={styles.subTitle}>Branch Highlights</h2>
        <div style={styles.imageGrid}>
          {/* Branch 1 */}
          <div style={styles.branchCard}>
            <Image
              src="/images/branchopol.png"
              alt="Opol Branch"
              layout="responsive"
              width={500}
              height={300}
              style={styles.highlightImage}
            />
            <p style={styles.branchDescription}>
              <strong>Grand Opening - Opol Branch</strong>
              <br />
              Join us in celebrating the grand opening of our Opol branch.
              Enjoy our 3-for-100 promo!
            </p>
          </div>

          {/* Branch 2 */}
          <div style={styles.branchCard}>
            <Image
              src="/images/branchdasmarinas.png"
              alt="Dasmarinas Branch"
              layout="responsive"
              width={500}
              height={300}
              style={styles.highlightImage}
            />
            <p style={styles.branchDescription}>
              <strong>Dasmarinas Branch</strong>
              <br />
              Our Dasmariñas branch conducted an outreach program to give back to the community.
            </p>
          
            </div>
            {/* Branch 3 */}
          <div style={styles.branchCard}>
            <Image
              src="/images/branchlingayen.png" // Replace with your actual image path
              alt="Lingayen Branch"
              layout="responsive"
              width={500}
              height={300}
              style={styles.highlightImage}
            />
            <p style={styles.branchDescription}>
              <strong>Lingayen Branch</strong>
              <br />
              Get ready, coffee lovers! Buhisan branch is opening soon!!
            </p>
          </div>

          {/* Branch 4 */}
          <div style={styles.branchCard}>
            <Image
              src="/images/branchopol.png" // Replace with your actual image path
              alt="Pol Branch"
              layout="responsive"
              width={500}
              height={300}
              style={styles.highlightImage}
            />
            <p style={styles.branchDescription}>
              <strong>Opol Branch</strong>
              <br />
              'Join us in celebrating the grand opening of our Opol branch. Enjoy our 3-for-100 promo!
            </p>
          </div>
          <div style={styles.branchCard}>
            <Image
              src="/images/branchtabunok.png" // Replace with your actual image path
              alt="Pol Branch"
              layout="responsive"
              width={500}
              height={300}
              style={styles.highlightImage}
            />
            <p style={styles.branchDescription}>
              <strong>Tabunok</strong>
              <br />
              Grand opening celebration with special guest Ariel Alegado. Don’t miss out!          
            </p>
          </div>
          <div style={styles.branchCard}>
            <Image
              src="/images/branchbuhisan.png" // Replace with your actual image path
              alt="Pol Branch"
              layout="responsive"
              width={500}
              height={300}
              style={styles.highlightImage}
            />
            <p style={styles.branchDescription}>
              <strong>Buhisan Branch</strong>
              <br />
              Get ready, coffee lovers! Buhisan branch is opening soon!            
              </p>
          </div>
        </div>
        
      </main>
    </div>
  );
}

const styles = {
  container: {
    fontFamily: "'Segoe UI', Tahoma, Geneva, Verdana, sans-serif",
    backgroundColor: "#f7f0e3",
    minHeight: "100vh",
    display: "flex",
    flexDirection: "row",
  },
  title: {
    fontSize: "3rem",
    fontWeight: "bold",
    textAlign: "center",
    color: "#4b3025", // Coffee-themed color
    marginBottom: "20px",
    textTransform: "uppercase",
    letterSpacing: "3px",
    textShadow: "2px 2px 4px rgba(0, 0, 0, 0.2)", // Subtle shadow effect
    padding: "10px 0",
    borderBottom: "2px solid #d49d6e", // Add a golden coffee-colored underline
    display: "inline-block",
    marginTop: "20px",
  },
  mainContent: {
    flex: 1,
    padding: "20px",
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
  },
  subTitle: {
    fontSize: "2rem",
    color: "#4b3025",
    marginBottom: "20px",
    fontWeight: "bold",
  },
  imageGrid: {
    display: "grid",
    gridTemplateColumns: "repeat(2, 1fr)",
    gap: "20px",
    width: "100%",
    maxWidth: "1200px",
  },
  branchCard: {
    backgroundColor: "#fff8e7",
    borderRadius: "12px",
    overflow: "hidden",
    boxShadow: "0px 4px 8px rgba(0, 0, 0, 0.1)",
    padding: "15px",
    textAlign: "center",
  },
  branchDescription: {
    marginTop: "10px",
    fontSize: "1rem",
    color: "#4b3025",
    fontWeight: "bold",
  },
  highlightImage: {
    borderRadius: "12px",
  },
};
